### ISSUE TITLE
How to decode deeply nested part of json using msgspec

### ISSUE THREAD

**Issue by giridhart** [jcrist/msgspec#583]
### Question

What is the correct way to decode part of json file?
Is there a way to do this without defining class Stuct for every level?
Also how to handle when the keys have a `.` like `test.name` ?

For example,
```
{
"test.name" : [
...
]
}
```

**Comment by FHU-yezi** [comment]
For the second question, you can use the `nane` argument in the `field` function.

**Comment by wikiped** [comment]
@giridhart

You may want to have a look at [Renaming Fileds](https://jcristharif.com/msgspec/structs.html#renaming-fields). And in particular at __"On the struct definition"__ part, which states that `rename` parameter can be:

  > - A mapping from field names to the renamed names...
  > - A callable (signature `rename(name: str) -> Optional[str])` ...

So if the builtin types are sufficient - the root `Struct` like this should work:

```python
class Root(msgspec.Struct, rename=lambda name: name.replace('_', '.')):
    ...

**Comment by giridhart** [comment]
Yes, now I remember reading about renaming fields, I should read few times to cement my memory. Thank you.
